docs/common/widget.ldoc

```diff
@@ -39,3 +39,64 @@
 --- Set/get a widget's buttons.
 -- @param _buttons The table of buttons that should bind to the widget.
 -- @function buttons
+
+
+--- When the layout (size) change.
+-- @signal widget::layout_changed
+
+--- When the widget content changed.
+-- @signal widget::redraw_needed
+
+--- When a mouse button is pressed over the widget.
+-- @signal button::press
+-- @tparam table widget The widget
+-- @tparam number lx The horizontal position.
+-- @tparam number ly The vertical position.
+-- @tparam number button The button number.
+-- @tparam table mods The modifiers (mod4, mod1 (alt), Control, Shift)
+-- @tparam table geometry
+-- @tparam number geometry.x The vertical position
+-- @tparam number geometry.y The horizontal position
+-- @tparam number geometry.width The widget
+-- @tparam number geometry.height The height
+-- @tparam drawable geometry.drawable The `drawable`
+-- @see mouse
+
+--- When a mouse button is released over the widget.
+-- @signal button::release
+-- @tparam table widget The widget
+-- @tparam number lx The horizontal position.
+-- @tparam number ly The vertical position.
+-- @tparam number button The button number.
+-- @tparam table mods The modifiers (mod4, mod1 (alt), Control, Shift)
+-- @tparam table geometry
+-- @tparam number geometry.x The vertical position
+-- @tparam number geometry.y The horizontal position
+-- @tparam number geometry.width The widget
+-- @tparam number geometry.height The height
+-- @tparam drawable geometry.drawable The `drawable`
+-- @see mouse
+
+--- When the mouse enter a widget.
+-- @signal mouse::enter
+-- @tparam table widget The widget
+-- @tparam table mods The modifiers (mod4, mod1 (alt), Control, Shift)
+-- @tparam table geometry
+-- @tparam number geometry.x The vertical position
+-- @tparam number geometry.y The horizontal position
+-- @tparam number geometry.width The widget
+-- @tparam number geometry.height The height
+-- @tparam drawable geometry.drawable The `drawable`
+-- @see mouse
+
+--- When the mouse leave a widget.
+-- @signal mouse::leave
+-- @tparam table widget The widget
+-- @tparam table mods The modifiers (mod4, mod1 (alt), Control, Shift)
+-- @tparam table geometry
+-- @tparam number geometry.x The vertical position
+-- @tparam number geometry.y The horizontal position
+-- @tparam number geometry.width The widget
+-- @tparam number geometry.height The height
+-- @tparam drawable geometry.drawable The `drawable`
+-- @see mouse
```

Comment:
Sorry, but we/I am inconsistent. :-)

enter/leave only have a single argument which is the result of `:find_widget` which is what you call `geometry`here. There are no modifiers (and I don't think there should be...)